src/ConnectedMode/UI/ProjectSelection/ProjectSelectionViewModel.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.ObjectModel;
+using SonarLint.VisualStudio.Core.WPF;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI.ProjectSelection;
+
+public class ProjectSelectionViewModel : ViewModelBase, IBindingStatusListener
+{
+    public event EventHandler BindingFinished;
+    public ObservableCollection<ServerProject> ProjectResults { get; } = [];
+
+    public ConnectionInfo.Connection Connection { get; }
+
+    public string ProjectSearchTerm
+    {
+        get => projectSearchTerm;
+        set
+        {
+            projectSearchTerm = value;
+            SearchForProject();
+            RaisePropertyChanged();
+        }
+    }
+
+    public ServerProject SelectedProject
+    {
+        get => selectedProject;
+        set
+        {
+            selectedProject = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsBindButtonEnabled));
+        }
+    }
+
+    public bool IsBindButtonEnabled => !BindClicked && SelectedProject != null;
+
+    public string BindingStatusText
+    {
+        get => bindingStatusText;
+        private set
+        {
+            bindingStatusText = value;
+            RaisePropertyChanged();
+        }
+    }
+    
+    public bool BindClicked
+    {
+        get => bindClicked;
+        private set
+        {
+            bindClicked = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsBindButtonEnabled));
+        }
+    }
+    
+    private string projectSearchTerm;
+    private ServerProject selectedProject;
+    private bool bindClicked;
+    private string bindingStatusText;
+
+    public ProjectSelectionViewModel(ConnectionInfo.Connection connection)
+    {
+        Connection = connection;
+    }
+
+    public void InitializeProjects(List<ServerProject> projects)
+    {
+        ProjectResults.Clear();
+        projects.ForEach(x => ProjectResults.Add(x));
+    }
+
+    public void Bind()
+    {
+        BindClicked = true;
```

Comment:
Please make sure to set the BindClicked to false. I know it is a dummy component, but if we want to demo it we should really be able to show how it would work.

You could add a delay, for example, as well